orttraining/orttraining/python/training/onnxblock/_graph_utils.py

```diff
@@ -58,7 +61,7 @@ def build_gradient_graph(accessor, user_args_requiring_grad, user_args_not_requi
     graph_inputs = model.graph.input
     initializers = []
     for initializer in model.graph.initializer:
-        if not initializer.name.startswith("onnx::"):
+        if not initializer.name.startswith("onnx::") and initializer.name not in quant_params:
```

Comment:
what does onnx:: refer to?

Reply:
This is a hack (and not a very good one at that).
- To build the gradient graph, we need to know which onnx model initializer is a "pytorch model parameter" and which is constant initializers that the exporter introduced.
- This is needed primarily to know which initializers need their gradient computed.
- There is no good way of knowing this from the exported onnx model.
- The way I do it here is that I check if the exported onnx model has the prefix `onnx::`. If it does, then this was introduced by the exporter and hence is not a pytorch model parameter. So, no need to build gradient for this initializer.
- While this is a hack, I have requested a feature to the exporter team to add information in the onnx model to be able to tell which initializer is a model parameter and which is not.